test/RibbonThetaVaultYearn.js

```diff
@@ -0,0 +1,3531 @@
+          "Cap exceeded"
+        );
+      });
+    });
+
+    describe("#setWithdrawalFee", () => {
+      it("reverts when not manager", async function () {
+        await expect(
+          this.vault.connect(userSigner).setWithdrawalFee(parseEther("0.1"))
+        ).to.be.revertedWith("Only manager");
+      });
+
+      it("reverts when withdrawal fee is 0", async function () {
+        await expect(
+          this.vault.connect(managerSigner).setWithdrawalFee(0)
+        ).to.be.revertedWith("withdrawalFee != 0");
+      });
+
+      it("reverts when withdrawal fee set to 30%", async function () {
+        await expect(
+          this.vault.connect(managerSigner).setWithdrawalFee(parseEther("30"))
+        ).to.be.revertedWith("withdrawalFee >= 30%");
+      });
+
+      it("sets the withdrawal fee", async function () {
+        const res = await this.vault
+          .connect(managerSigner)
+          .setWithdrawalFee(parseEther("0.1"));
+
+        await expect(res)
+          .to.emit(this.vault, "WithdrawalFeeSet")
+          .withArgs(parseEther("0.005"), parseEther("0.1"));
+
+        assert.equal(
+          (await this.vault.instantWithdrawalFee()).toString(),
+          parseEther("0.1").toString()
+        );
+      });
+    });
+
+    describe("#currentOptionExpiry", () => {
+      it("should return 0 when currentOption not set", async function () {
+        assert.equal((await this.vault.currentOptionExpiry()).toString(), "0");
+      });
+    });
+
+    describe("#decimals", () => {
+      it("should return 18 for decimals", async function () {
+        assert.equal(
+          (await this.vault.decimals()).toString(),
+          this.tokenDecimals.toString()
+        );
+      });
+    });
+  });
+}
+
+async function signOrderForSwap({
+  counterpartyAddress,
+  vaultAddress,
+  sellToken,
+  buyToken,
+  sellAmount,
+  buyAmount,
+  signerPrivateKey,
+}) {
+  let order = createOrder({
+    signer: {
+      wallet: counterpartyAddress,
+      token: buyToken,
+      amount: buyAmount,
+    },
+    sender: {
+      wallet: vaultAddress,
+      token: sellToken,
+      amount: sellAmount,
+    },
+    affiliate: {
+      wallet: TRADER_AFFILIATE,
+    },
+  });
+
+  const signedOrder = await signTypedDataOrder(
+    order,
+    signerPrivateKey,
+    SWAP_CONTRACT
+  );
+  return signedOrder;
+}
+
+async function depositIntoVault(asset, vault, amount) {
+  if (asset === WETH_ADDRESS) {
+    await vault.depositETH({ value: amount });
+  } else {
+    await vault.deposit(amount);
+  }
+}
+
+async function setupYieldToken(
+  addressToDeposit,
+  assetContract,
+  collateralAsset,
+  vault,
+  amount
+) {
+  for (let i = 0; i < addressToDeposit.length; i++) {
+    const weth = assetContract.connect(addressToDeposit[i]);
+    await weth.deposit({ value: parseEther("10") });
+    const yvtoken = (
+      await getContractAt("IYearnVault", collateralAsset)
+    ).connect(addressToDeposit[i]);
```

Comment:
Idea: One way to make the tests more realistic is to make the `pricePerShare` > 1*10**18

Reply:
the pricePerShare is greater than 1*10**18. an example: 1010936530067422878 (~1.0109 eth per one yvweth)

Reply:
Oh sorry, you used the mainnet yvWETH address so it has a realistic `pricePerShare`. I had the impression that you used a mocked version.